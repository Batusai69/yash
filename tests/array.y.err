array: index 100 is out of range (the actual size of array $aaa is 11)
array: index -100 is out of range (the actual size of array $aaa is 11)
===== 1 =====
array: `--no-such-option' is not a valid option
array: $aaa is read-only
array: $aaa is read-only
array: $aaa is read-only
array: $aaa is read-only
array: $aaa is read-only
array: $aaa is read-only
array: $aaa is read-only
