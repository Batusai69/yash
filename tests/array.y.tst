# array.y.tst: yash-specific test of the array builtin
# vim: set ft=sh ts=8 sts=4 sw=4 noet:

ary=(test of array)
array aaa 1 2 3
echo 1 $?
array
echo 2 $?
array aaa 1 2 3 4 5 6 7 8 9
echo 3 $? $aaa
array -d aaa 6 2 10 2 8
echo 4 $? $aaa
array -d aaa -1 -5 -20 0
echo 5 $? $aaa
array -i aaa 0 0
echo 6 $? $aaa
array -i aaa 2 2 3
echo 7 $? $aaa
array -i aaa 6 6
echo 8 $? $aaa
array -i aaa -1 8
echo 9 $? $aaa
array -i aaa 100 9
echo 10 $? $aaa
array -i aaa -100 !
echo 11 $? $aaa
array -s aaa 1 0
array -s aaa 2 1
array -s aaa 10 -
echo 12 $? $aaa
array -s aaa 100 x || array -s aaa -100 x || echo 13 $? $aaa

echo ===== 1 =====
echo ===== 1 ===== >&2

array --no-such-option
echo array no-such-option $?
(array >&- 2>/dev/null)
echo array output error $?

readonly aaa
array aaa
echo array readonly 1 $?
array aaa 1 2 3
echo array readonly 2 $?
array -d aaa
echo array readonly 3 $?
array -d aaa 1 2
echo array readonly 4 $?
array -i aaa 3
echo array readonly 5 $?
array -i aaa 3 1 2
echo array readonly 6 $?
array -s aaa 2 -
echo array readonly 7 $?
echo $aaa
