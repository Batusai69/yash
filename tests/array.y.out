1 0
aaa=('1' '2' '3')
ary=('test' 'of' 'array')
2 0
3 0 1 2 3 4 5 6 7 8 9
4 0 1 3 4 5 7 9
5 0 1 4 5 7
6 0 0 1 4 5 7
7 0 0 1 2 3 4 5 7
8 0 0 1 2 3 4 5 6 7
9 0 0 1 2 3 4 5 6 7 8
10 0 0 1 2 3 4 5 6 7 8 9
11 0 ! 0 1 2 3 4 5 6 7 8 9
12 0 0 1 1 2 3 4 5 6 7 - 9
13 1 0 1 1 2 3 4 5 6 7 - 9
===== 1 =====
array no-such-option 2
array output error 1
array readonly 1 1
array readonly 2 1
array readonly 3 1
array readonly 4 1
array readonly 5 1
array readonly 6 1
array readonly 7 1
0 1 1 2 3 4 5 6 7 - 9
